### PR TITLE
Hide processing context

### PR DESCRIPTION
**Pull request by wicknicks**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java [hunk 1]
```diff
@@ -491,10 +490,10 @@ ErrorHandlingMetrics errorHandlingMetrics(ConnectorTaskId id) {
         return new ErrorHandlingMetrics(id, metrics);
     }
 
-    private ProcessingContext sinkProcessingContext(ConnectorTaskId id, ConnectorConfig connConfig,
+    private List<ErrorReporter> sinkProcessingContext(ConnectorTaskId id, ConnectorConfig connConfig,
```
Comment (mageshn) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java:493]: method name needs to be changed

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java [hunk 2]
```diff
@@ -506,18 +505,18 @@ private ProcessingContext sinkProcessingContext(ConnectorTaskId id, ConnectorCon
             reporters.add(reporter);
         }
 
-        return new ProcessingContext(reporters);
+        return reporters;
     }
 
-    private ProcessingContext sourceProcessingContext(ConnectorTaskId id, ConnectorConfig connConfig,
+    private List<ErrorReporter> sourceProcessingContext(ConnectorTaskId id, ConnectorConfig connConfig,
```
Comment (mageshn) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java:511]: method name :)

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/errors/RetryWithToleranceExecutor.java [hunk 3]
```diff
@@ -95,6 +98,8 @@
     private RetryWithToleranceExecutorConfig config;
     private ErrorHandlingMetrics errorHandlingMetrics;
 
+    protected ProcessingContext context = new ProcessingContext();
```
Comment (mageshn) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/errors/RetryWithToleranceExecutor.java:101]: can processing context be just an inner class?

**Review by mageshn — COMMENTED** [review]
had few minor comments

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java (hunk 1 above)
Comment (wicknicks) [reply connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java:493]: fixed in the main PR.

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java (hunk 2 above)
Comment (wicknicks) [reply connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java:511]: fixed in the main PR.

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/errors/RetryWithToleranceExecutor.java (hunk 3 above)
Comment (wicknicks) [reply connect/runtime/src/main/java/org/apache/kafka/connect/runtime/errors/RetryWithToleranceExecutor.java:101]: it's a package private class right now. any advantages of nesting?

**Pull request closed without merging**